### PR TITLE
Define SDWebImageDownloader convenience method

### PR DESCRIPTION
**Pull request by iliaskarim**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [ ] I have added the required tests to prove the fix/feature I am adding
* [ ] I have updated the documentation (if necessary)
* [ ] I have run the tests and they pass
* [ ] I have run the lint and it passes (`pod lib lint`)

### Pull Request Description

Based on the discussion on #2611 it seems reasonable to contribute a convenience method for downloading images.

### CODE REVIEW

File: SDWebImage.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dreampiggy) [inline SDWebImage.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Please don't submit this auto generated files into the repo. You can ignore this from git-config. Or maybe you can modify the `.gitignore` files to add this line of file into it.

Comment (iliaskarim) [reply SDWebImage.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Ok, thanks @dreampiggy. If you'd like I can open a separate PR to add this file or ignore it. My understanding is files in xcshareddata are intended to be version controlled.

**Comment by iliaskarim** [comment]
@dreampiggy please re-review at your leisure.

File: SDWebImage.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 1 above)
Comment (dreampiggy) [reply SDWebImage.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Typically you're right. But this `IDEWorkspaceChecks.plist` from Xcode 9.3 is a exception. This files only contains the 32-bit warning for Xcode now. So it's useless for most projects.

If we want to use it, I can submit this files again. Sorry for bother.

**Comment by dreampiggy** [comment]
Seems OK...

Are you using Swift with SDWebImage ? Actually the downloader is not a high-frequency used components, so I think why the downlaoder API have all args here is for user who care about the detail args behavior.

But anyway, it's not hard to maintain the code after this API was introduced.

**Comment by iliaskarim** [comment]
No, @dreampiggy we are not using Swift yet. I appreciate your thoughts. There have been a couple of times that this interface could have come in handy. Please merge at your discretion.

**Comment by dreampiggy** [comment]
So I merge it ?...

Though it's a new API, but I think this one is just a convient method (Swift optiona arg), which will not make future refactory become more complicated.

**Pull request merged**